src/wasmopcodecnt.c

```diff
@@ -0,0 +1,192 @@
+};
+WASM_STATIC_ASSERT(NUM_FLAGS == WASM_ARRAY_SIZE(s_options));
+
+static void on_option(struct WasmOptionParser* parser,
+                      struct WasmOption* option,
+                      const char* argument) {
+  switch (option->id) {
+    case FLAG_VERBOSE:
+      s_verbose++;
+      wasm_init_file_writer_existing(&s_log_stream_writer, stdout);
+      wasm_init_stream(&s_log_stream, &s_log_stream_writer.base, NULL);
+      s_read_binary_options.log_stream = &s_log_stream;
+      break;
+
+    case FLAG_HELP:
+      wasm_print_help(parser, PROGRAM_NAME);
+      exit(0);
+      break;
+
+    case FLAG_OUTPUT:
+      s_outfile = argument;
+      break;
+
+    case FLAG_USE_LIBC_ALLOCATOR:
+      s_use_libc_allocator = WASM_TRUE;
+      break;
+  }
+}
+
+static void on_argument(struct WasmOptionParser* parser, const char* argument) {
+  s_infile = argument;
+}
+
+static void on_option_error(struct WasmOptionParser* parser,
+                            const char* message) {
+  WASM_FATAL("%s\n", message);
+}
+
+static void parse_options(int argc, char** argv) {
+  WasmOptionParser parser;
+  WASM_ZERO_MEMORY(parser);
+  parser.description = s_description;
+  parser.options = s_options;
+  parser.num_options = WASM_ARRAY_SIZE(s_options);
+  parser.on_option = on_option;
+  parser.on_argument = on_argument;
+  parser.on_error = on_option_error;
+  wasm_parse_options(&parser, argc, argv);
+
+  if (!s_infile) {
+    wasm_print_help(&parser, PROGRAM_NAME);
+    WASM_FATAL("No filename given.\n");
+  }
+}
+
+#define V(rtype, type1, type2, mem_size, code, NAME, text) [code] = text,
+static const char* s_opcode_name[] = {WASM_FOREACH_OPCODE(V)};
+#undef V
+
+int main(int argc, char** argv) {
+  WasmResult result;
+  WasmStackAllocator stack_allocator;
+  WasmAllocator* allocator;
+
+  wasm_init_stdio();
+  parse_options(argc, argv);
+
+  if (s_use_libc_allocator) {
+    allocator = &g_wasm_libc_allocator;
+  } else {
+    wasm_init_stack_allocator(&stack_allocator, &g_wasm_libc_allocator);
+    allocator = &stack_allocator.allocator;
+  }
+
+  void* data;
+  size_t size;
+  result = wasm_read_file(allocator, s_infile, &data, &size);
+  if (WASM_SUCCEEDED(result)) {
+    size_t wasm_opcode_count[WASM_ARRAY_SIZE(s_opcode_name)];
+    WASM_ZERO_MEMORY(wasm_opcode_count);
+    result = wasm_read_binary_opcnt(
+        allocator, data, size, &s_read_binary_options, &s_error_handler,
+        wasm_opcode_count, (size_t)WASM_ARRAY_SIZE(wasm_opcode_count));
+    FILE* out = stdout;
+    if (s_outfile) {
+      out = fopen(s_outfile, "w");
+      if (!out)
+        ERROR("fopen \"%s\" failed, errno=%d\n", s_outfile, errno);
+      result = WASM_ERROR;
+    }
+    if (WASM_SUCCEEDED(result)) {
+      for (size_t i = 0; i < WASM_ARRAY_SIZE(wasm_opcode_count); ++i) {
+        if (wasm_opcode_count[i] == 0)
+          continue;
+        fprintf(out, "%s: %" PRIuMAX "\n", s_opcode_name[i],
+                (uintmax_t)wasm_opcode_count[i]);
```

Comment:
I typically use `PRIzd` for this (although I suppose those should all be `PRIzu`) and keep it as a `size_t`.